components/help/components/formatting.jsx

```diff
@@ -5,81 +5,851 @@ import React from 'react';
+                    id='help.formatting.links.description'
+                    defaultMessage='Create labeled links by putting the desired text in square brackets and the associated link in normal brackets.'
+                />
+            </p>
+            <span className='codespan__pre-wrap'>
+                <code>
+                    {'['}
+                    <FormattedMessage
+                        id='help.formatting.linkEx'
+                        defaultMessage='Check out Mattermost!'
+                    />
+                    {'](https://about.mattermost.com/)'}
+                </code>
+            </span>
+            <p/>
+            <p>
+                <FormattedMessage
+                    id='help.formatting.renders'
+                    defaultMessage='Renders as:'
+                />
+                {' '}
+                <span>
+                    <a href='https://about.mattermost.com/'>
+                        <FormattedMessage
+                            id='help.formatting.linkEx'
+                            defaultMessage='Check out Mattermost!'
+                        />
+                    </a>
+                </span>
+            </p>
+            <h2 className='markdown__heading'>
+                <FormattedMessage
+                    id='help.formatting.images.title'
+                    defaultMessage='In-line Images'
+                />
+            </h2>
+            <p>
+                <FormattedMessage
+                    id='help.formatting.images.description'
+                    defaultMessage='Create in-line images using an {exclamationMarkSign} followed by the alt text in square brackets and the link in normal brackets. Add hover text by placing it in quotes after the link.'
+                    values={{
+                        exclamationMarkSign: <span className='codespan__pre-wrap'><code>{'!'}</code></span>,
+                    }}
+                />
+            </p>
+            <div className='post-code post-code--wrap'>
+                <code className='hljs'>
+                    <FormattedMessage
+                        id='help.formatting.imagesExample'
+                        defaultMessage={'![alt text](link \\"hover text\\")\\n\\nand\\n\\n[![Build Status](https://travis-ci.org/mattermost/mattermost-server.svg?branch=master)](https://travis-ci.org/mattermost/mattermost-server) [![Github](https://assets-cdn.github.com/favicon.ico)](https://github.com/mattermost/mattermost-server)'}
+                    />
+                </code>
+            </div>
+            <p>
+                <FormattedMessage
+                    id='help.formatting.renders'
+                    defaultMessage='Renders as:'
+                />
+            </p>
+            <p>
+                <FormattedMessage
+                    id='help.formatting.imagesExample.altText'
+                    defaultMessage='alt text'
+                >
+                    {(altText) => (
```

Comment:
This is really cool. I've never seen this way of using `FormattedMessage` before. We might be able to use this in other places to get rid of `localizeMessage` without using `injectIntl` like we do elsewhere